Microsoft.Toolkit.Uwp.Connectivity/BluetoothLEHelper/ObservableGattDeviceService.cs

```diff
@@ -147,10 +147,9 @@ protected virtual void OnPropertyChanged([CallerMemberName] string propertyName
         }
 
         /// <summary>
-        /// Gets all the characteristics of this service
+        /// Populate the characteristics in <see cref="Characteristics"/>.
         /// </summary>
-        /// <returns>The status of the communication with the GATT device.</returns>
-        private async Task<GattCommunicationStatus> GetAllCharacteristics()
+        private async void PopulateAllCharacteristicsAsync()
```

Comment:
Move back to Task returning. We shouldn't have async voids if they can be avoided

Reply:
They should but here we are in a fire-and-forget case. Having an `async void` or discarding the returned `Task` instance will have the same behavior.

Reply:
This isn't about if it is or isn't fire and forget. This is about what happens when those methods being awaited throw an exception. Replace the call to Service.GetCharacteristicsAsync to a call to a new method defined below.

```
        private IAsyncOperation<GattCharacteristicsResult> GetCharacteristicsAsync(BluetoothCacheMode cacheMode)
        {
            throw new NotImplementedException();
        }
```
Run the application with the async void and what happens?